packages/abilities/src/types.ts

```diff
@@ -63,24 +63,26 @@ export interface Ability {
 	output_schema?: Record< string, any >;
 
 	/**
-	 * Callback function for client-side abilities.
-	 * If present, the ability will be executed locally in the browser.
-	 * If not present, the ability will be executed via REST API on the server.
+	 * Callback function for ability execution.
+	 * This property is required for all abilities.
 	 */
 	callback?: AbilityCallback;
 
 	/**
-	 * Client Permission callback for abilities.
+	 * Permission callback for abilities.
 	 * Called before executing the ability to check if it's allowed.
 	 * If it returns false, the ability execution will be denied.
 	 */
 	permissionCallback?: PermissionCallback;
 
 	/**
 	 * Metadata about the ability.
+	 *
 	 * @see WP_Ability::get_meta()
 	 */
 	meta?: {
+		_clientRegistered?: boolean;
+		_serverRegistered?: boolean;
 		annotations?: {
```

Comment:
Can we remove "annotations" it's kind of the same thing as "meta", so let's just flatten meta